source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst

```diff
@@ -1305,6 +1407,20 @@ recovering" error::
         if is_shutdown(code) or (error was from <4.2):
             clear connection pool for server
 
+    def shouldHandleError(td, response):
+        currentServer = td.servers[server.address]
+        currentGeneration = currentServer.pool.generation
+        generation = get connection generation from response
+        if generation < currentGeneration:
+            # Stale generation number.
+            return False
+
+        currentTv = currentServer.topologyVersion
+        if currentTv is null or "topologyVersion" not in response:
+            return True
+        # True if the current server's topologyDescription is < error’s
+        return compareTopologyVersion(currentTv, response["topologyVersion"]) < 0
```

Comment:
`compareTopologyVersion` is no longer defined anywhere, should we reintroduce the pseudo code for this?